documentation/specs/src/economics/public-goods-funding.md

```diff
@@ -83,9 +61,34 @@ The `--spending-cap` argument is a `float` $0 < x \leq 1$, which indicates the m
 
 A council consisting of the same members should also be able to propose multiple spending caps (with the same multisig address). These will be voted on as separate councils and votes counted separately.
 
+Proposing candidacy as a PGF council is something that is done at any time. This simply signals to the rest of Governance that a given established multisignature account address is willing to be voted on during a PGF council election in the future.
+
+Candidacy proposals last a default of 31 epochs. There is no limit to the number of times a council can be proposed for candidacy. This helps ensure that no PGF council is elected that does not intend to become one.
```

Comment:
I think we can either use a `Map<Council, (Epoch, Url)>` or `VecDeque<(Council, Epoch, Url)>` where `Council = (Address, spending_cap)` to store the candidates in storage.

At each epoch, we can remove from this collection the expired candidacies. If we use the queue, the voters can use the indexes of the queue to specify their choices (even though this might be more error prone

Reply:
What is the benefit of using the queue? What do you mean by indexes of the queue to specify choices?

Reply:
We can use a `Vec` as well, with a queue we might be faster in certain operations (like pruning from the head). A voter can use the index of the council in the vector (or queue) instead of providing the tuple `(Address, spending_cap)`. At the moment I'm more concerned of what happens if some candidates are dropped in the middle of a voting period

Reply:
If we remove expired candidacies during a voting period indexes would point to another council so this is not safe. At this point we can use the `Map` I'd say